### PR TITLE
Add Pentax K-3 II and fix lens model detection for K-3(II)

### PR DESCRIPTION
**Pull request by tomtor**
Cleanup of

https://github.com/darktable-org/darktable/pull/929

### CODE REVIEW

File: src/common/exif.cc [hunk 1]
```diff
@@ -572,6 +572,11 @@ static bool dt_exif_read_exif_data(dt_image_t *img, Exiv2::ExifData &exifData)
     {
       dt_strlcpy_to_utf8(img->exif_lens, sizeof(img->exif_lens), pos, exifData);
     }
+    else if((pos = exifData.findKey(Exiv2::ExifKey("Exif.PentaxDng.LensType"))) != exifData.end()
```
Comment (pedrocr) [inline src/common/exif.cc]: This will break on older exiv2 versions. It probably needs a version check like we do with Exif.CanonFi.FocusDistanceUpper.

**Comment by pedrocr** [comment]
You don't need to open new PRs when you make changes. If you are just adding new commits you can just push and if you are rebasing or starting over you can just push with --force to replace what is here.

File: src/common/exif.cc (hunk 1 above)
Comment (tomtor) [inline src/common/exif.cc]: Thanks, should be fixed now...

File: src/common/exif.cc [hunk 2]
```diff
@@ -572,6 +572,12 @@ static bool dt_exif_read_exif_data(dt_image_t *img, Exiv2::ExifData &exifData)
     {
       dt_strlcpy_to_utf8(img->exif_lens, sizeof(img->exif_lens), pos, exifData);
     }
+    else if(Exiv2::testVersion(0,25,0)
```
Comment (pedrocr) [inline src/common/exif.cc]: Did you check that this will only be available in the 0.25? If 0.24 already has it we can use that instead.

Comment (tomtor) [inline src/common/exif.cc]: Yes, I did. 0.24 is from 02-Dec-2013 and the specific patch is from 8 months ago and has as target 0.25

Comment (pedrocr) [inline src/common/exif.cc]: Cool, thanks

File: src/external/rawspeed/data/cameras.xml [hunk 3]
```diff
@@ -5585,6 +5585,17 @@
...
+        <Camera make="RICOH IMAGING COMPANY, LTD." model="PENTAX K-3 II">
+                <ID make="Pentax" model="K-3 II">PENTAX K-3 II</ID>
+                <CFA width="2" height="2">
+                        <Color x="0" y="0">RED</Color>
+                        <Color x="1" y="0">GREEN</Color>
+                        <Color x="0" y="1">GREEN</Color>
+                        <Color x="1" y="1">BLUE</Color>
+                </CFA>
+                <Crop x="4" y="0" width="-36" height="0"/>
+                <Sensor black="1" white="15865"/>
```
Comment (pedrocr) [inline src/external/rawspeed/data/cameras.xml]: I think you mentioned different ISOs having different black or white points. That can also be set, just see the Samsung NX1 for example. However judging by the color matrix the sensor is the same as the original K-3 and we didn't do that for that one so maybe it's not really relevant. How big are the differences?

Comment (tomtor) [inline src/external/rawspeed/data/cameras.xml]: iminimal: black/white is 1/15865 for iso 800, 1/15866 (iso 1600),  2/15865 (iso 3200)

Comment (pedrocr) [inline src/external/rawspeed/data/cameras.xml]: Ok, just ignore it then. Seems ready to merge then.

**Comment by pmjdebruijn** [comment]
Thanks!

**Pull request merged**